src/flamenco/runtime/program/fd_bpf_loader_v3_program.c

```diff
@@ -1492,7 +1503,15 @@ process_loader_upgradeable_instruction( fd_exec_instr_ctx_t * instr_ctx ) {
     if( FD_UNLIKELY( PROGRAMDATA_METADATA_SIZE>programdata_account->meta->dlen ) ) {
       return FD_EXECUTOR_INSTR_ERR_ACC_DATA_TOO_SMALL;
     }
-    err = deploy_program( instr_ctx, programdata_account->data + PROGRAMDATA_METADATA_SIZE, fd_ulong_sat_add( SIZE_OF_PROGRAM, new_len ) );
+
+    uchar * programdata_data = programdata_account->data + PROGRAMDATA_METADATA_SIZE;
+    ulong   programdata_size = new_len                   - PROGRAMDATA_METADATA_SIZE;
+
+    /* We use programdata_size instead of UpgradeableLoaderState::size_of_program().saturating_add(programdata_len)
+       because that is the length of our buffer, and that argument to deploy_program (account_size) in the rust code is not used
+       in our version of deploy_program because we do not have a program cache
+       https://github.com/anza-xyz/agave/blob/da6f7f2d317ed978701b08c3cc3e8d62756e565a/programs/bpf_loader/src/lib.rs#L146 */
+    err = deploy_program( instr_ctx, programdata_data, programdata_size );
```

Comment:
This seems to be an incorrect permalink

Reply:
I removed it and moved the comment to `deploy_program`